server/src/main/java/org/candlepin/service/impl/DefaultEntitlementCertServiceAdapter.java

```diff
@@ -449,6 +449,10 @@ public String cleanUpPrefix(String contentPrefix) throws IOException {
 
         log.debug("WE HAVE {} POOL QUANTITIES TO LOOP THROUGH", poolQuantities.size());
 
+        // Serials need to be saved to get generated ID.
+        log.debug("Persisting new certificate serials");
+        serialCurator.saveOrUpdateAll(serialMap.values(), false, false);
+
```

Comment:
This change is required to fix one of Junit. I am not sure whether it is fine to change this Adapter class or not.

Reply:
I stepped through the code and this change is fine. Amusingly, there is a comment a few lines up on 446 which calls out the original logic and why this wasn't done initially.

In addition to moving this block, that comment should be removed as well, as it is no longer accurate.

Reply:
Done.